### ISSUE TITLE
Support adding user location in jail step

### ISSUE THREAD

**Issue by darrenvong** [Couchers-org/couchers#423]
Not really sure if I should call it a bug since it's a recent addition to the jail step, but as a result, the current React frontend offers no way for a new user to add a location to the profile, which means they can't escape jail.

We currently ask the user's location during the sign up form, but it's not provided in the lat-lng format required by
```
rpc SetLocation(SetLocationReq) returns (JailInfoRes) {
    // Set the user's location
    // If you're not jailed, do this with api.UpdateProfile!
  }
```
Also it doesn't call the jail API to update the user's location.

I guess we should probably remove this in the sign up form and follow the same format as the alpha and have this as another jail step, since that field alone wouldn't allow the user to set a radius.

**Comment by lucaslcode** [comment]
I am working on this currently. It's taking a while to fight with the map library :D

**Comment by darrenvong** [comment]
@lucaslcode Just had a glance at the code... did you have come up with all those trigonometry maths?!! Lol that stuff is gonna be fun to fix if there are any weird bugs in...

**Comment by lucaslcode** [comment]
Some but not all! Not much to be done unfortunately. I tried to use the inbuilt `project` and `unproject` functions as much as possible.

**Comment by darrenvong** [comment]
Maybe a link to where you found those calculations or comment on how you come up with it would help in case we need to fix it in the future

**Comment by lucaslcode** [comment]
Closed in #427
